### PR TITLE
Support mixed `INT8` + `FP16` in one model

### PR DESCRIPTION
**Pull request by yiliu30**
## Type of Change

feature
API changed or not

## Description
- [x] Support convert unquantized `linear` into `fp16`
- [ ] Extend the fp16 ops list to align with https://pytorch.org/docs/stable/amp.html#cpu-ops-that-can-autocast-to-bfloat16 in a separate PR

### Usage
```python
model = export(model, example_inputs=example_inputs)

quant_config = get_default_static_config()
quant_config.set_local(torch.nn.Linear, StaticQuantConfig(w_dtype="fp16", act_dtype="fp16"))
# prepare
prepare_model = prepare(model, quant_config)
# calibrate
for i in range(2):
    prepare_model(*example_inputs)
# convert
converted_model = convert(prepare_model)

```

## How has this PR been tested?
Pre-CI
Some extension test later.

## Dependency Change?
expecttest

### CODE REVIEW

**Review by zehao-intel — APPROVED** [review]

**Comment by yiliu30** [comment]
> Shall we add logger.info to teach user that ipex doesn't support fp16? `quant_config.set_local(torch.nn.Linear, StaticQuantConfig(w_dtype="fp16", act_dtype="fp16"))`

Thanks for your suggestions, will refine it by a separate PR.

File: test/3x/torch/quantization/test_pt2e_quant.py [hunk 1]
```diff
@@ -145,3 +168,62 @@ def test_prepare_and_convert_on_llm(self, force_not_import_ipex):
...
+                expected_node_occurrence.get(node, 0) == cnt
+            ), f"Node {node} should occur {cnt} times, but {node_in_graph[node]}"
+
+        # inference
+        from torch._inductor import config
+
+        config.freezing = True
+        opt_model = torch.compile(converted_model)
+        out = opt_model(*example_inputs)
+        assert out is not None
```
Comment (Kaihui-intel) [inline test/3x/torch/quantization/test_pt2e_quant.py:229]: Suggest adding accuracy check.

**Review by Kaihui-intel — APPROVED** [review]

File: test/3x/torch/quantization/test_pt2e_quant.py (hunk 1 above)
Comment (yiliu30) [reply test/3x/torch/quantization/test_pt2e_quant.py:234]: Thanks, added.
